lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,49 @@
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+
+			// TODO: render generator initial view
+			this.textContent = this.store.state.example;
+
+			this.onStateChange = this.handleStateChange.bind(this);
+
+
+			// TODO: add click event
+			this.addEventListener('click', () => {
+				this.store.dispatch({
+					type: 'BUY_GENERATOR',
+					payload: 'Bought Generator'
```

Comment:
payload should be reading from the generator name. This can only buy a static generator.